core/src/main/java/org/elasticsearch/action/bulk/BulkItemResponse.java

```diff
@@ -90,9 +101,39 @@ public RestStatus getStatus() {
             return this.status;
         }
 
+        /**
+         * The actual cause of the failure.
+         */
         public Throwable getCause() {
             return cause;
         }
+
+        @Override
+        public Failure readFrom(StreamInput in) throws IOException {
+            return new Failure(in.readString(), in.readString(), in.readOptionalString(), in.readThrowable());
```

Comment:
Filed as an [issue]
(https://github.com/elastic/elasticsearch/issues/17086) so I'll do it on the master branch and backport it rather than just do it here.